### PR TITLE
Support size options on builtin volumes

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1982164

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@mheon PTAL

**Comment by mheon** [comment]
Nice and simple, LGTM

File: libpod/volume.go [hunk 1]
```diff
@@ -49,6 +49,8 @@ type VolumeConfig struct {
 	UID int `json:"uid"`
 	// GID the volume will be created as.
 	GID int `json:"gid"`
+	// Size maximum of the volume.
+	Size int64 `json:"gid"`
```
Comment (Luap99) [inline libpod/volume.go:53]: Should we store this as uint64 given that `SetQuota()` accepts uint64?
Also the json field name is wrong `s/gid/size/`.

File: libpod/runtime_volume_linux.go [hunk 2]
```diff
@@ -106,6 +107,26 @@ func (r *Runtime) newVolume(ctx context.Context, options ...VolumeCreateOption)
...
+		}
+		quota := quota.Quota{}
+		if volume.config.Size > 0 || volume.config.Inodes > 0 {
+			if !projectQuotaSupported {
+				return nil, errors.New("Volume options size and inodes not supported. Filesystem does not support Project Quota")
+			}
+			quota.Size = volume.config.Size
+			quota.Inodes = volume.config.Inodes
+		}
+		if projectQuotaSupported {
```
Comment (mtrmac) [inline libpod/runtime_volume_linux.go:124]:
```suggestion
```
Just like this? We don’t need to set quota (and potentially conflict with XFS project IDs).

Comment (mtrmac) [reply libpod/runtime_volume_linux.go:124]: don’t need to set quota … if there isn’t a quota request.

Comment (mtrmac) [reply libpod/runtime_volume_linux.go:124]: … actually, all of this, including the quota support check, can be wrapped inside the `volume.config.{Size,Inodes} > 0` condition.

/me steps away to regain focus.

**Comment by rhatdan** [comment]
@giuseppe @vrothberg @TomSweeneyRedHat @saschagrunert @baude @jwhonce PTAL

**Review by giuseppe — APPROVED** [review]
LGTM

**Review by saschagrunert — APPROVED** [review]
LGTM

**Comment by TomSweeneyRedHat** [comment]
@mheon @rhatdan, unless I missed it, it doesn't look like this was cherry-picked into the v3.3 branch.  Can we make that so, so we get this bug fixed in RHEL 8.5?

**Comment by mheon** [comment]
We're feature-frozen at this point, and this counts as a feature. Will that cause us problems?

**Comment by TomSweeneyRedHat** [comment]
@mheon it has an associated BZ, so I think it's fair game.

**Pull request merged**